src/components/Questionnaire/QuestionnaireForm.tsx

```diff
@@ -77,10 +78,19 @@ export function QuestionnaireForm({
     prefetch: !!questionnaireSlug && !FIXED_QUESTIONNAIRES[questionnaireSlug],
   });
 
-  const { mutate: submitBatch, isProcessing } = useMutation(
-    routes.batchRequest,
-    { silent: true },
-  );
+  const { mutate: submitBatch, isPending: isProcessing } = useMutation({
+    mutationFn: mutate(routes.batchRequest),
+    onSuccess: () => {
+      toast.success(t("questionnaire_submitted_successfully"));
+      onSubmit?.();
+    },
+    onError: (error: any) => {
```

Comment:
Let's not use `any` type. You may however, cast it inside the fn. by `error.cause as unknown as <Expected Type>` if needed.

Also are you sure the `error` object had `results`? I believe it'd be wrapped in `HttpError` error object which has to be accessed via `cause`.

Do test it out.

```suggestion
    onError: (error) => {
```

Reply:
Ah yes, I guess you're right about both points,